Plugins/CGAL/vtkCGAL/vtkCGALEfficientRANSAC.cxx

```diff
@@ -0,0 +1,277 @@
+* 
+* \brief todo
+* 
+* 
+* 
+* Inputs: todo
+* Output: todo
+* 
+*/
+
+#include <vtkCGALEfficientRANSAC.h>
+
+// -- VTK
+#include <vtkCellData.h>
+#include <vtkCGALUtilities.h>
+#include <vtkFieldData.h>
+#include <vtkFloatArray.h>
+#include <vtkInformation.h>
+#include <vtkInformationVector.h>
+#include <vtkIntArray.h>
+#include <vtkPointData.h>
+#include <vtkSmartPointer.h>
+#include <vtkTimerLog.h>
+
+// -- CGAL
+#include <CGAL/property_map.h>
+#include <CGAL/Point_with_normal_3.h>
+#include <CGAL/Exact_predicates_inexact_constructions_kernel.h>
+
+#include <CGAL/Shape_detection/Efficient_RANSAC.h>
+
+// Type declarations.
+typedef CGAL::Exact_predicates_inexact_constructions_kernel                     Kernel;
+typedef Kernel::FT                                                              FT;
+typedef CGAL::First_of_pair_property_map<vtkCGALUtilities::Point_with_normal0>  Point_map;
+typedef CGAL::Second_of_pair_property_map<vtkCGALUtilities::Point_with_normal0> Normal_map;
+
+typedef CGAL::Shape_detection::Efficient_RANSAC_traits
+<Kernel, vtkCGALUtilities::Pwn_vector0, Point_map, Normal_map>  Traits;
+typedef CGAL::Shape_detection::Efficient_RANSAC<Traits>         Efficient_ransac;
+typedef CGAL::Shape_detection::Plane<Traits>                    Plane;
+
+// -----------------------------------------------------------------------------
+// Declare the plugin
+vtkStandardNewMacro(vtkCGALEfficientRANSAC);
+
+// -----------------------------------------------------------------------------
+// Constructor
+// Todo.
+vtkCGALEfficientRANSAC::vtkCGALEfficientRANSAC()
+{
+  this->NumberOfIterations = 1;
+  this->UseParameters = 0;
+  this->MinPoints = 200;
+
+  this->Probability = 0.05;
+  this->Epsilon = 0.002;
+  this->ClusterEpsilon = 0.01;
+  this->MaxNormalDeviation = 0.9;
+}
+
+// -----------------------------------------------------------------------------
+// Todo.
+int vtkCGALEfficientRANSAC::RequestData(
+  vtkInformation *vtkNotUsed(request),
+  vtkInformationVector **inputVector,
+  vtkInformationVector *outputVector)
+{
+  // Get the info objects
+  vtkInformation *inInfo = inputVector[0]->GetInformationObject(0);
+  vtkInformation *outInfo = outputVector->GetInformationObject(0);
+
+  // Get the input and output
+  vtkPolyData *input = vtkPolyData::SafeDownCast(
+    inInfo->Get(vtkDataObject::DATA_OBJECT()));
+  vtkPolyData *output = vtkPolyData::SafeDownCast(
+    outInfo->Get(vtkDataObject::DATA_OBJECT()));
+
+  output->CopyStructure(input);
+  output->GetCellData()->PassData(input->GetCellData());
+  output->GetFieldData()->PassData(input->GetFieldData());
+  output->GetPointData()->PassData(input->GetPointData());
+
+  // Points with normals.
+  vtkCGALUtilities::Pwn_vector0 points;
+
+  if (!vtkCGALUtilities::vtkPolyDataToPointsWithNormals(input, points))
+  {
+    vtkErrorMacro("Failed to convert input mesh. Make sure to supply points normals.");
+    return 0;
+  }
+
+  std::cout << "Efficient RANSAC" << std::endl;
```

Comment:
Is this required? Is it just for debugging?

Reply:
no, good point